tests/test_entity_ranking.py

```diff
@@ -0,0 +1,116 @@
+        score = torch.tensor([score]).expand_as(s)
+        return score
+
+    def score_sp(self, s: Tensor, p: Tensor, o: Tensor = None) -> Tensor:
+        score = torch.tensor([self.SCORE])
+        score = score.expand(len(s), len(o))
+        return score
+
+    def score_po(self, p: Tensor, o: Tensor, s: Tensor = None) -> Tensor:
+        score = torch.tensor([self.SCORE])
+        score = score.expand(len(o), len(s))
+        return score
+
+    def score_sp_po(self, s: Tensor, p: Tensor, o: Tensor, entity_subset: Tensor = None) -> Tensor:
+        num_entities = self.dataset.num_entities()
+        if entity_subset is None:
+            entity_subset = torch.arange(num_entities)
+
+        a = self.score_sp(s, p, entity_subset)
+        b = self.score_po(p, o, entity_subset)
+        return torch.cat((a, b), dim=1)
+
+    def prepare_job(self, job: "Job", **kwargs):
+        pass
+
+
+class BaseTestEntityRanking:
+
+    def __init__(self, spo_different, options={}):
+        self.spo_different = spo_different
+        self.dataset_name = "dataset_test"
+        self.options = options
+
+    def setUp(self):
+        config = create_config(self.dataset_name)
+        config.set_all(self.options)
+        config.set_all({"eval.trace_level": "example"})
+        config.folder = "."
+        config.log_folder = tempfile.mkdtemp(prefix="kge-")
+
+        dataset = Dataset.create(
+            config, folder=get_dataset_folder(self.dataset_name)
+        )
+
+        model = TestModel(config, dataset, self.spo_different)
+        self.job = EntityRankingJob(config=config, dataset=dataset, parent_job=None, model=model)
+
+    def test_tie_handling(self):
+
+        # run the evaluation
+        self.job.run()
+
+        # since all entities get the same score, num_ties should equal num_entities, and rank should be 0
+        num_ties = self.job.dataset.num_entities()
+        rank = 0
+
+        # read trace file and check ranks are what they should be
+        trace_file = self.job.config.tracefile()
+        with open(trace_file, 'r') as file:
+            yaml_lines = file.readlines()
+
+        for yaml_line in yaml_lines:
+            yaml_line = yaml.safe_load(yaml_line)
+            if yaml_line['job_id'] == self.job.job_id \
+                    and 'event' in yaml_line.keys() and yaml_line['event'] == "example_rank":
+                self.assertEqual(self.job._get_ranks(rank, num_ties) + 1, yaml_line['rank'])
+
+
+class TestEntityRankingSPODifferent(BaseTestEntityRanking, unittest.TestCase):
+    def __init__(self, methodName="runTest"):
+        unittest.TestCase.__init__(self, methodName=methodName)
+        BaseTestEntityRanking.__init__(self, spo_different=True)
+
+
+class TestEntityRankingSPOSame(BaseTestEntityRanking, unittest.TestCase):
+    def __init__(self, methodName="runTest"):
+        unittest.TestCase.__init__(self, methodName=methodName)
+        BaseTestEntityRanking.__init__(self, spo_different=False)
```

Comment:
Why not use a single `TestEntityRanking` test with two methods `test_ties_exact` and `test_ties_close` or so?